components/create-post-page/SetPriceAndDescription.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  SimpleGrid,
+  Spacer,
+  Text,
+  Textarea,
+  VStack,
+} from "@chakra-ui/react";
+import { PopulatedBook } from "@lib/services/book";
+import { ChangeEvent, FormEvent, useState } from "react";
+import BookCard from "../BookCard";
+
+interface Props {
+  book: PopulatedBook;
+  onSubmitPressed: (description: string, price: number) => void;
+}
+
+export default function SetPriceAndDescription({
+  book,
+  onSubmitPressed,
+}: Props) {
+  const [description, setDescription] = useState("");
+  const [price, setPrice] = useState(0);
+  const [valid, setValid] = useState(false);
+  const [sending, setSending] = useState(false);
+
+  const handleSubmit = (event: FormEvent) => {
+    if (!isNaN(price) && isFinite(price) && price >= 0) {
+      setSending(true);
+      onSubmitPressed(description, price);
+    }
+    event.preventDefault();
+  };
+
+  const handlePriceChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newPriceString = e.target.value.trim();
+    const newPriceAsFloat = Number.parseFloat(newPriceString);
+    if (
+      !isNaN(newPriceAsFloat) &&
+      isFinite(newPriceAsFloat) &&
+      newPriceAsFloat >= 0
+    ) {
+      setPrice(newPriceAsFloat);
+      setValid(true);
+    } else {
+      setValid(false);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <VStack spacing={7}>
+        <BookCard book={book} isLinkActive={false} />
+        <SimpleGrid
+          columns={2}
+          spacing={3}
+          templateColumns="1fr 4fr"
+          style={{ alignItems: "center" }}
+        >
+          {/* Row 1 */}
+          <Box>
+            <Text as="b">Description:</Text>
+          </Box>
+          <Box>
+            <Textarea
+              backgroundColor="secondaryBackground"
+              placeholder="Describe the condition of the book"
+              onChange={(e) => setDescription(e.target.value.trim())}
+            />
+          </Box>
+
+          {/* Row 2 */}
+          <Box>
+            <Text as="b">Asking Price:</Text>
+          </Box>
+          <Box>
+            <InputGroup>
+              <InputLeftElement pointerEvents="none" color="gray.300">
+                $
+              </InputLeftElement>
+              <Input
+                backgroundColor="secondaryBackground"
+                placeholder="Price"
+                onChange={(e) => handlePriceChange(e)}
+                type="number"
+              ></Input>
+            </InputGroup>
+          </Box>
+
+          {/* Row 3 */}
+          <Box>{/* empty */}</Box>
+          <Box>
+            <Flex direction="column">
+              <Box paddingBottom={2}>
+                <Text color="gray.500" textAlign="right">
+                  {/* TODO: implement backend to serve this information */}
+                  {book.campusStorePrice
+                    ? `Sold new for $${book.campusStorePrice / 100}`
```

Comment:
Might want to call `.toFixed(2)` just to be safe (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed)